internal/pkg/clients/kite.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2025 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clients
```

Comment:
I'd suggest rename this package to `kite` with code in `internal/pkg/kite`. Then rename the struct and API:
- KiteClient -> Client
- NewKiteClient -> NewClient